File: p3/Deepspace/Loot.rb
#encoding:utf-8
require_relative "LootToUI.rb"
module Deepspace
    class Loot
        def initialize(_nSupplies,_nWeapons,_nShields,_nHangars,_nMedals)
            @nSupplies = _nSupplies
            @nWeapons = _nWeapons
            @nShields = _nShields
            @nHangars = _nHangars
            @nMedals = _nMedals
        end
        def getNSupplies()
            @nSupplies
        end 
        def getNWeapons()
            @nWeapons
        end 
        def getNShields()
            @nShields
        end 
        def getNHangars()
            @nHangars
        end 
        def getNMedals()
            @nMedals
        end 
        def to_s()
            return "Supplies: #{@nSupplies}, Weapons: #{@nWeapons}, Shields: #{@nShields}, Hangars: #{@nHangars}, Medals: #{@nMedals}"
        end 
        def getUIversion()
			Deepspace::LootToUI.new(self)
		end

    end
    #a = Loot.new(1,2,3,4,5)
    #puts a.to_s
end

File: p4/deepspace/src/deepspace/Loot.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author Sergio
 */
public class Loot {
    private final int nSupplies;
    private final int nWeapons;
    private final int nShields;
    private final int nHangars;
    private final int nMedals;
    private boolean getEfficient;
    private boolean spaceCity;
    
    Loot(int Supplies, int Weapons, int Shields, int Hangars, int Medals){
        nSupplies = Supplies;
        nWeapons = Weapons;
        nShields = Shields;
        nHangars = Hangars;
        nMedals = Medals;
        getEfficient = false;
        spaceCity = false;
    }
    
    Loot(int Supplies, int Weapons, int Shields, int Hangars, int Medals, boolean ef, boolean city){
        nSupplies = Supplies;
        nWeapons = Weapons;
        nShields = Shields;
        nHangars = Hangars;
        nMedals = Medals;
        getEfficient = ef;
        spaceCity = city;
    }    
    
    int getNSupplies(){
        return nSupplies;
    }
    
    int getNWeapons(){
        return nWeapons;   
    }
    
    int getNShields(){
        return nShields;
    }
    
    int getNHangars(){
        return nHangars;
    }
    
    int getNMedals(){
        return nMedals;
    }
    
    boolean getEfficient(){
        return getEfficient;
    }
    
    boolean spaceCity(){
        return spaceCity;
    }
    
    LootToUI getUIversion(){
        return new LootToUI(this);
    }
}
